doc/source/examples/extended_examples/sfem/sfem.py

```diff
@@ -0,0 +1,758 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import math
+import random
+from typing import Callable, Tuple
+
+import numpy as np
+
+
+def find_solution(
+    func: Callable[[float], float],
+    derivative_func: Callable[[float], float],
+    acceptable_solution_error: float,
+    solution_range: Tuple[float, float],
+) -> float:
+    """Find the solution of g(x) = 0 within solution range where g(x) is non-linear.
+
+    Parameters
+    ----------
+    func : Callable[float, float]
+        The function definition
+    derivative_func : Callable[float, float]
+        The derivative of the above function
+    acceptable_solution_error : float
+        Error at which the solution is acceptable
+    solution_range : Tuple[float, float]
+        The range within which the solution will be searched
+
+    Returns
+    -------
+    float
+        Solution to g(x) = 0
+    """
+
+    current_guess = random.uniform(*solution_range)
+    iteration_counter = 1
+
+    while True:
+        if iteration_counter > 100:
+            iteration_counter = 1
+            current_guess = random.uniform(*solution_range)
+            continue
+
+        updated_guess = current_guess - func(current_guess) / derivative_func(
+            current_guess
+        )
+        error = abs(updated_guess - current_guess)
+
+        if error < acceptable_solution_error and not (
+            solution_range[0] < updated_guess < solution_range[1]
+        ):
+            current_guess = random.uniform(*solution_range)
+            continue
+        elif error < acceptable_solution_error and (
+            solution_range[0] < updated_guess < solution_range[1]
+        ):
+            return updated_guess
+
+        current_guess = updated_guess
+        iteration_counter += 1
+
+
+def evaluate_KL_cosine_terms(
+    domain: Tuple[float, float], correl_length_param: float, min_eigen_value: float
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Build array of eigenvalues and constants of the cosine terms in the KL expansion of a gaussian stochastic process.
```

Comment:
```suggestion
    """Build array of eigenvalues and constants of the cosine terms in the KL expansion of a Gaussian stochastic process.
```
The term "Gaussian" is capitalized because it is derived from the name of the mathematician Carl Friedrich Gauss.